### PR TITLE
Patch 10

### PR DESCRIPTION
**Pull request by PowerGamer1**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | yes |
| Tests pass? | yes |
| Fixed issues | https://github.com/yiisoft/yii2/issues/12073 |

### CODE REVIEW

File: framework/widgets/ActiveField.php [hunk 1]
```diff
@@ -315,10 +317,17 @@ public function error($options = [])
      *
      * @return $this the field object itself
      */
-    public function hint($content, $options = [])
+    public function hint($content = null, $options = [])
```
Comment (samdark) [inline framework/widgets/ActiveField.php]: In order for it to be merged in 2.0 it should not change public method signature.

Comment (PowerGamer1) [inline framework/widgets/ActiveField.php]: What is the problem with this particular change? It brings the method signature in line with `label()` method and does not break BC in any way.

Comment (rob006) [inline framework/widgets/ActiveField.php]: @PowerGamer1 It will break any class that override `hint()` method using old signature: https://3v4l.org/IoSqL

Comment (PowerGamer1) [inline framework/widgets/ActiveField.php]: Alright, that would break BC indeed. I added upgrade instructions to UPGRADE.md.

Comment (rob006) [inline framework/widgets/ActiveField.php]: But why you change this signature? There is no big benefits.

Comment (PowerGamer1) [inline framework/widgets/ActiveField.php]: @rob006 
As I said, for consistency with `public function label($label = null, $options = [])`. IMO you either do something properly or don't do it at all. Besides, affected code upgrade for this change is super easy.

Comment (samdark) [inline framework/widgets/ActiveField.php]: I think I'll merge it w/o signature change in 2.0 and w/ signature change in 2.1.

**Comment by tunecino** [comment]
@PowerGamer1 please check the [Yii Versioning Policy](https://github.com/yiisoft/yii2/blob/master/docs/internals/versions.md) section. someone somewhere will upgrade a production app to 2.0.10 and his online code may stop working properly. Yii should guarantee that no changes is required for minor releases upgrades.

**Comment by samdark** [comment]
Merged differently to both 2.0 and 2.1. Thank you for contributing!

**Pull request closed without merging**